app/controllers/katello/api/v2/content_view_filters_controller.rb

```diff
@@ -82,6 +82,49 @@ def destroy
       respond_for_show :resource => @filter
     end
 
+    api :put, "/content_views/:content_view_id/filters/:id/remove_filter_rules", N_("bulk delete filter rules")
+    api :put, "/content_view_filters/:id/remove_filter_rules", N_("bulk delete filter rules")
+    param :content_view_id, :number, :desc => N_("content view identifier")
+    param :id, :number, :desc => N_("filter identifier"), :required => true
+    param :rule_ids, Array, of: :number, :desc => N_("filter identifiers"), :required => true
+    def remove_filter_rules
+      rule_clazz = ContentViewFilter.rule_class_for(@filter)
+      rule_clazz.where(id: params[:rule_ids]).try(:destroy_all)
+      respond_for_show(resource: @filter)
+    end
+
+    api :put, "/content_views/:content_view_id/filters/:id/add_filter_rules", N_("bulk add filter rules")
+    api :put, "/content_view_filters/:id/add_filter_rules", N_("bulk add filter rules")
+    param :content_view_id, :number, :desc => N_("content view identifier")
+    param :id, :number, :desc => N_("filter identifier"), :required => true
+    param :rules_params, Array, :desc => N_("Rules to be added"), :required => false
+    def add_filter_rules
+      rule_clazz = ContentViewFilter.rule_class_for(@filter)
+      rules = []
+      rules_params[:rules_params].each do |rule_params|
+        rules = (rule_params[:name] || []).map do |name|
+          rule_clazz.create!(rule_params.except(:name).merge(:filter => @filter, name: name))
+        end
+
+        rules += (rule_params[:errata_ids] || []).map do |errata_id|
+          rule_clazz.create!(rule_params.except(:errata_ids)
+                                 .merge(filter: @filter, errata_id: errata_id))
+        end
+        if rule_params[:module_stream_ids]
+          rules += (rule_params[:module_stream_ids] || []).map do |module_stream_id|
+            rule_clazz.create!(rule_params.except(:module_stream_ids)
+                                   .merge(filter: @filter, module_stream_id: module_stream_id))
+          end
+        end
```

Comment:
how many rules do you see being created at once?   I wonder if we should use active record import to do a bulk insert?

Reply:
I would say a maximum of 50 (max records per page) if a user selects everything on a page and adds, given multi page selection is not supported..I am not sure if people do that though with filter rules..I imagine only a few package groups added to a filter cause more than that would be a really complicated content view.

Reply:
i guess we can hold off for now, easy to add later on